### PR TITLE
Pushing schematisations to API

### PR DESCRIPTION
**Pull request by caspervdw**

### CODE REVIEW

File: threedi_model_migration/api_utils.py [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+            break
+        elif oa_revision.number > revision.revision_nr:
+            continue
+        if oa_revision.commit_date == revision.last_update:
+            return revision
+        else:
+            break
+
+
+def get_latest_revision(
```
Comment (jpprins1) [inline threedi_model_migration/api_utils.py:111]: Note: `/v3-beta/schematisations/{id}/latest_revision/` also exists.

File: threedi_model_migration/application.py [hunk 2]
```diff
@@ -342,6 +357,74 @@ def report(base_path: Path):
...
+            f"{repository} does not exist locally, please download it first."
+        )
+
+    inspection_path = base_path / INSPECTION_RELPATH / f"{slug}.plan.json"
+    with inspection_path.open("r") as f:
+        plan = json.load(f, object_hook=custom_json_object_hook)
+
+    schematisations: List[Schematisation] = plan["schematisations"]
+    with ThreediApi(env_file=env_file, version="v3-beta", asynchronous=False) as api:
+        api: V3BetaApi = api
```
Comment (jpprins1) [inline threedi_model_migration/application.py:381]: Note, you can just do:
`api: V3BetaApi`

File: threedi_model_migration/application.py [hunk 3]
```diff
@@ -342,6 +357,74 @@ def report(base_path: Path):
...
 
+def push(
+    base_path: Path,
+    slug: str,
+    mode: PushMode = PushMode.incremental,
+    env_file: Optional[Path] = None,
+    last_update: Optional[datetime] = None,
+):
+    """Aggregate all plans into 1 repository and 1 schematisation CSV"""
+    mode = PushMode(mode)
```
Comment (jpprins1) [inline threedi_model_migration/application.py:368]: mode already is a `PushMode` here. Could assert it.

**Review by jpprins1 — APPROVED** [review]
Looks very good! I need this locally to inject test models! ;)

File: threedi_model_migration/api_utils.py (hunk 1 above)
Comment (caspervdw) [reply threedi_model_migration/api_utils.py:111]: Yes, but that could include a commit done via the new route (while we want the latest commit from the modeldatabank)

**Pull request merged**